_posts/2022-06-22-domain-management-for-multitenant-app-in-aws.markdown

```diff
@@ -0,0 +1,62 @@
+
+It might have more business value for you to focus on maintaining and developing the application, but managing the domain is critical in keeping the application live and accessible.
+
+### Solution
+
+The diagram below is the solution to how you can typically start building it, but it will soon become challenging when it grows.
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureSingleAccount.png)
+
+Generally, it is a good practice to segregate the production environment from the development environment, which should also apply to domain management. This approach has no segregation between environments and hosts all domain names within a single location. It increases the risk of impacting all tenants when there are service disruptions. 
+
+The architecture diagram below shows how to manage domains for all your tenants in multiple environments.  
+
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureDiagram.png)
+
+AWS Route53 domain management provides the ability to segregate the management of domain names between accounts. DNS is a public-facing element you would want to minimize downtime as it will directly impact your revenue or reputation. Having segregation between accounts, especially for non-production and production, is key to reducing the risk of damage to production domains. It is also beneficial for segregating billing costs; hence, you will get better insights. 
+
+Besides, you can configure a single CloudFront distribution to serve requests from multiple origins or subdomains and an SSL certificate to encrypt the information. A single distribution and SSL certificate can reduce the overhead for maintenance, such as renewing all certificates for each subdomain. 
+
+#### Principle 
+
+There are a few principles you should follow, which are listed below:
+
+1. You should not provision resources under the top-level domains because your non-production and production account should be responsible for managing its subdomains and resources. 
+2. Manage the registration of top-level domains from a central account. It is to ensure you have the main point to manage all domain names and subdomains should all be in its workload account to keep it consistent. 
+3. All sub-domains for an application should live in the same workload account to reduce complexity. 
+
+Lastly, a single domain management stack that manages all the resources makes change management more straightforward. You can then achieve the automation by setting up a CI/CD pipeline for the stack. 
+
+
+#### Wrapping Up.
+
+ Setting up a robust pipeline for domain management is the first step in getting your multi-tenant cloud infrastructure right. 
```

Comment:
The summary doesn't really match the content of the article for me.

* Is the intent of the post to demonstrate (with an example/tutorial) how to build a pipeline for domain management?
* Or is it to introduce a pattern for managing multi-tenant domains?
* Or something else?

Any of the above is a great topic, but don't try to do everything.

When I get to the end of an article, I'm looking for (depending upon the type of article):
* An 'ah-ha' moment from your article that will make me think/reflect.  
* A follow on question that will make me want to dive into the topic further.
* The answer to my immediate problem